src/ploomber_core/validate.py

```diff
@@ -0,0 +1,111 @@
+import difflib
+from ploomber_core.exceptions import MissingKeysValidationError, ValidationError
+from ploomber_core.io import pretty_print
+import collections.abc
+
+
+def _is_set_or_sequence(obj) -> bool:
+    """
+    Checks if obj is a sequence or a set.
+    Returns False if obj is a string.
+    """
+    if isinstance(obj, str):
+        return False
+    if isinstance(obj, collections.abc.Sequence):
+        return True
+    if isinstance(obj, set):
+        return True
```

Comment:
if this is not a str nor a sequence/set, it'll return None, perhaps it's better to do:

```python
return isinstance(obj, (collections.abc.Sequence, set))
```

a lot shorter an the only two possible return values are True/False